assets/page/SettingsAccount.svelte

```diff
@@ -9,19 +9,36 @@ import {getContext} from 'svelte';
 import {l} from '../js/i18n';
 import {route} from '../store/Route';
 
-const colorSchemeOptions = [['auto', 'Auto'], ['dark', 'Dark'], ['light', 'Light']];
 const user = getContext('user');
 const updateUserOp = user.api.operation('updateUser');
 
+// "colorSchemeOptions" will get calculated each time "theme" changes
+$: colorSchemeOptions = calculateColorSchemeOptions(theme);
+$: colorScheme = resetColorScheme(theme);
+
+function calculateColorSchemeOptions(id) {
+  const options = [['auto', 'Auto']];
+  const theme = user.themes[id] || {variants: {}};
+  if (theme.variants.dark) options.push(['dark', 'Dark']);
+  if (theme.variants.light) options.push(['light', 'Light']);
+  return options;
+}
+
+function resetColorScheme(id) {
```

Comment:
I think this whole function is redundant: It assigns to `colorScheme`, but also returns `colorScheme` which earlier gets assigned to `colorScheme`. Maybe you would rather merge `calculateColorSchemeOptions()` and `resetColorScheme()`, like this:

```javascript
$: calculateColorSchemeOptions(theme);

function calculateColorSchemeOptions(id) {
  const options = [['auto', 'Auto']];
  const theme = user.themes[id] || {variants: {}};
  if (theme.variants.dark) options.push(['dark', 'Dark']);
  if (theme.variants.light) options.push(['light', 'Light']);
  colorSchemeOptions = options;
  colorScheme = 'auto';
}
```

Reply:
Done